src/test/scala/mesosphere/util/CapConcurrentExecutionsTest.scala

```diff
@@ -0,0 +1,119 @@
+package mesosphere.util
+
+import com.codahale.metrics.MetricRegistry
+import mesosphere.marathon.MarathonSpec
+import mesosphere.marathon.integration.setup.WaitTestSupport
+import mesosphere.marathon.metrics.Metrics
+import mesosphere.marathon.test.MarathonActorSupport
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.{ GivenWhenThen, Matchers }
+
+import scala.concurrent.duration._
+import scala.concurrent.{ Future, Promise }
+
+class CapConcurrentExecutionsTest extends MarathonActorSupport with MarathonSpec with Matchers with GivenWhenThen
+    with ScalaFutures {
+  def capMetrics: CapConcurrentExecutionsMetrics = new CapConcurrentExecutionsMetrics(
+    new Metrics(new MetricRegistry),
+    classOf[CapConcurrentExecutionsTest]
+  )
+
+  test("submit successful futures after each other") {
+    val serialize = CapConcurrentExecutions(capMetrics, system, "serialize1", maxParallel = 1, maxQueued = 10)
+    try {
+      val result1 = serialize(Future.successful(1)).futureValue
```

Comment:
To be honest - I don't see the point of those tests: you basically block and wait for the future to succeed. Proposal:
```scala
      var count = 0
      serialize(Future{ Thread.sleep(50); count += 1; count should be(1) })
      serialize(Future{ count += 1; count should be(2) })
      serialize(Future{ count += 1; count should be(3) })
```